src/app/plans/change/page.tsx

```diff
@@ -0,0 +1,97 @@
+'use client'
+
+import { usePlan } from "@/hooks"
+import { useW3, DID } from "@w3ui/react"
+import { CheckCircleIcon } from '@heroicons/react/24/outline'
+import DefaultLoader from "@/components/Loader"
+import { useState } from "react"
+
+interface PlanSectionProps {
+  planID: DID
+  planName: string
+  flatFee: number
+  flatFeeAllotment: number
+  perGbFee: number
+}
+
+const planRanks: Record<string, number> = {
+  'did:web:starter.web3.storage': 0,
+  'did:web:lite.web3.storage': 1,
+  'did:web:business.web3.storage': 2
+}
+
+const buttonText = (currentPlan: string, newPlan: string) => (planRanks[currentPlan] > planRanks[newPlan]) ? 'Downgrade' : 'Upgrade'
+
+
+function PlanSection ({ planID, planName, flatFee, flatFeeAllotment, perGbFee }: PlanSectionProps) {
+  const [{ accounts }] = useW3()
+  const { data: plan, setPlan, isLoading } = usePlan(accounts[0])
+  const currentPlanID = plan?.product
+  const isCurrentPlan = currentPlanID === planID
+  const [isUpdatingPlan, setIsUpdatingPlan] = useState(false)
+  async function selectPlan (selectedPlanID: DID) {
+    setIsUpdatingPlan(true)
+    await setPlan(selectedPlanID)
+    setIsUpdatingPlan(false)
+  }
+  return (
+    <div className={`flex flex-col items-center rounded border-2 border-solid border-slate-800 w-96 p-8 ${isCurrentPlan ? 'bg-blue-800/10' : 'bg-slate-800/10'}`}>
```

Comment:
Yea I'm open to this! I have found this pattern is great for components that I intend to be reusable (ie, stuff that you'd stick in a "design system" library or something) and kind of clunky without many real benefits for more page-specific stuff.

I also think it's worth considering developing a good pattern for mocking SWR-based data loading components like this rather than trying to get everything into the properties - sometimes it just feels way cleaner for a component to load its own state rather than try to pass state down through multiple levels of components.

TLDR I agree and/but also think there's some nuance here based on past experiences trying to make this happen - let's dig into it soon!